xbmc/pvr/providers/PVRProviders.cpp

```diff
@@ -107,14 +107,13 @@ bool CPVRProviders::LoadFromDatabase(const std::vector<std::shared_ptr<CPVRClien
   const std::shared_ptr<CPVRDatabase> database = CServiceBroker::GetPVRManager().GetTVDatabase();
   if (database)
   {
-    bool bChanged = false;
-
     CPVRProviders providers;
     database->Get(providers, clients);
 
     for (auto& provider : providers.GetProvidersList())
     {
-      bChanged |= (CheckAndAddEntry(provider, ProviderUpdateMode::BY_DATABASE) != nullptr);
+      if (provider)
```

Comment:
Why this check? Can it actually happen that provider is nullptr?

Reply:
By the definition of `CheckAndAddEntry()`, yes it’s possible. In this particular case I don’t think so but likely easier reading to stick to the function return values.

Reply:
Not sure I understand. CheckAndAddEntry can return nullptr, but how could a nullptr-CPVRProvider find its way into the providers list? BTW, the original code does not check that, right?

Reply:
Fair point. Will clean that up.

Reply:
Cleanup up.